amqp-impl/src/main/java/io/streamnative/pulsar/handlers/amqp/AmqpChannel.java

```diff
@@ -302,7 +302,7 @@ public void receiveExchangeDelete(AMQShortString exchange, boolean ifUnused, boo
             } else {
                 if (amqpExchange.getDurable()) {
                     PersistentTopic topic = (PersistentTopic) amqpExchange.getTopic();
-                    if (ifUnused && topic.getSubscriptions().isEmpty()) {
+                    if (ifUnused && amqpExchange.getQueueSize() > 0) {
```

Comment:
It seems that AmqpExchange availability has nothing to do with AmqpExchange durable, maybe we could do the availability check before judge AmqpExchange durable.